### PR TITLE
feat: move ci-prow namespace to smaug

### PR DESCRIPTION
**Pull request by mimotej**
Moving namespace of ci-prow from zero cluster to smaug.
Related: https://github.com/operate-first/argocd-apps/pull/186

### CODE REVIEW

File: cluster-scope/overlays/prod/moc/smaug/kustomization.yaml [hunk 1]
```diff
@@ -6,6 +6,8 @@ resources:
   - ../common
   - ../../../../base/config.openshift.io/oauths/cluster
   - ../../../../base/core/namespaces/acme-operator
+    - ../../../../base/core/namespaces/opf-ci-prow
```
Comment (harshad16) [inline cluster-scope/overlays/prod/moc/smaug/kustomization.yaml:9]:
```suggestion
   - ../../../../base/core/namespaces/opf-ci-prow
```

File: cluster-scope/overlays/prod/moc/smaug/kustomization.yaml [hunk 2]
```diff
@@ -6,6 +6,8 @@ resources:
   - ../common
   - ../../../../base/config.openshift.io/oauths/cluster
```
Comment (harshad16) [inline cluster-scope/overlays/prod/moc/smaug/kustomization.yaml:7]:
```suggestion
  - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/prowjobs.prow.k8s.io
  - ../../../../base/config.openshift.io/oauths/cluster
```

**Review by harshad16 — COMMENTED** [review]
lgtm
just few minor comments

File: cluster-scope/overlays/prod/moc/smaug/kustomization.yaml [hunk 3]
```diff
@@ -4,8 +4,11 @@ kind: Kustomization
 
 resources:
   - ../common
+  - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/prowjobs.prow.k8s.io
   - ../../../../base/config.openshift.io/oauths/cluster
   - ../../../../base/core/namespaces/acme-operator
+   - ../../../../base/core/namespaces/opf-ci-prow
+  - ../../../../base/core/namespaces/opf-ci-test-pods
```
Comment (harshad16) [inline cluster-scope/overlays/prod/moc/smaug/kustomization.yaml:11]:
```suggestion
  - ../../../../base/core/namespaces/opf-ci-prow
  - ../../../../base/core/namespaces/opf-ci-test-pods
```

**Review by harshad16 — APPROVED** [review]
lgtm
thanks

**Review by 4n4nd — APPROVED** [review]
/lgtm

**Pull request merged**